lms/security.py

```diff
@@ -173,6 +179,10 @@ def _get_lti_user(request):
         partial(bearer_token_schema.lti_user, location="form"),
         OAuthCallbackSchema(request).lti_user,
     ]
+    if "id_token" in request.params:
```

Comment:
I'm a bit confused over the main motivation here. Avoiding extra work seems like a good reason to do this, but the comment says it's because we might have a bogus token?

I'm not quite sure what this means in the current (quite strange) design where we just try everything until something works. In that design presumably we'd expect it to be broken and just move onto the next thing?

Reply:
It's probably not worded very well.

What I mean is that I just don't outright do `schemas = [LTI13AuthSchema]` just in case id_token is there but somehow doesn't work and we'd like to continue checking other schemas. Not sure how probable that scenario is.